document-store/src/main/java/org/hypertrace/core/documentstore/query/AllSelection.java

```diff
@@ -0,0 +1,20 @@
+package org.hypertrace.core.documentstore.query;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class AllSelection implements Selection {
```

Comment:
Thinking about this again - we probably don't need to support "all". Expecting specific fields should be good enough.

Reply:
Yeah. With respect to MongoDB, selecting specific-fields is sufficient. Since Postgres (which is an SQL) is also supported as of now and for any future use cases, I think, it is good to have.

Since these APIs are not bound to a specific database, I wanted to keep it as generic as possible so that we don't break the compatibility frequently.

What do you think?

Reply:
Yeah, this isn't specific to a particular DB. It's whether our query interface needs to support "all" fields or not. The existing query interface also expects the user to specify the required fields upfront. In the new Query class, the top fields seem to have a mix of expressions, definitions, and now the selection interface. Was thinking if we can at least simplify the selection one by removing the * support and look into the other two. We can merge this PR and take the open comments from Ronak as well in your next.

Reply:
Sure. Will work on it in the next PR.